lua/telescope/config.lua

```diff
@@ -187,12 +187,32 @@ local telescope_defaults = {
     - "absolute"  display absolute paths
     - "shorten"   only display the first character of each directory in
                   the path
+    - "function"  use a custom function defined in path_display_custom
 
     path_display can also be set to 'hidden' string to hide file names
 
     Default: {}]]
   },
 
+  path_display_custom = { nil, [[
+    Custom function for formatting the Path Display. Function called when path_display is set to 'function'
+
+    Example function that formats the path to "file.ext (full path)":
+
+        path_display_custom = function(opts, filepath)
+            iterator = string.gmatch(filepath, "(.-)([^\\/]-%.?([^%.\\/]*))$")
+            result = filepath
+            for key, value in iterator do
+                if key ~= '' then
+                    result = value .. " (" .. key .. ")"
+                end
+            end
+            return result
+        end
```

Comment:
I think that this function is probably a bit too complicated to serve as an example in the docs.
Is there another function that you could can think of that is simpler but still useful in this context?

Reply:
Yeah... I see what you're saying. It can be as simple as:

path_display = function(opts, filepath)
-- custom formatting logic
return formattedPath
end

I'll see if I can think of something a bit more useful but not overly complex. Suggestions always welcome :)

Reply:
Just saying but we have a `path_tail` util function so we could use that
```lua
path_display = function(opts, path)
  local tail = require("telescope.utils").path_tail(path)
  return string.format("%s (%s)", tail, path)
end
```

Just tried it works like a charm. So we are just working on finalizing the documentation. We have docgen in ci, if ci doesn't wanna work i run it locally and push the docgen changes to this branch when its done :)

Reply:
Nice! Much better and works fine on Windows as well. This would be a good example for the docs I would think. Simple enough.